src/spring-cloud/azext_spring_cloud/tanzu.py

```
@@ -235,6 +239,46 @@ def tanzu_app_deploy(cmd, client, resource_group, service, name, artifact_path=N
     return client.deployments.deploy(resource_group, service, name, deployment_name, build_result_id=build_result_id)
 
 
+def tanzu_app_tail_log(cmd, client, resource_group, service, name, deployment=None, instance=None,
+                       follow=False, lines=50, since=None, limit=2048):
+    '''tanzu_app_tail_log
+    Get real time logs from the app.
+    Throw exception if app or deployment not found.
+    '''
+    deployment = _assert_deployment_exist_and_retrieve(cmd, client, resource_group, service, name)
+    instances = deployment.properties.instances
+    if len(instances) > 1:
+        logger.warning("Multiple app instances found:")
+        for temp_instance in instances:
+            logger.warning("{}".format(temp_instance.name))
+        logger.warning("Please use '-i/--instance' parameter to specify the instance name")

```

Throw CLIError instead of logger.warning and return None?